drop-transfer/src/ws/server/mod.rs

```diff
@@ -412,11 +413,17 @@ async fn handle_client(
 }
 
 impl FileXferTask {
-    pub fn new(file: crate::File, xfer: crate::Transfer, location: PathBuf) -> crate::Result<Self> {
+    pub fn new(
+        file: crate::File,
+        xfer: crate::Transfer,
+        location: PathBuf,
+        base_dir: PathBuf,
```

Comment:
Isn't the location the same thing as base_dir ?

Reply:
location here is the already glued together final path of the file

Reply:
`full_location` or `absolute_path` are better names to describe `location`

Reply:
Yes, the naming could be changed for a more descriptive one

Reply:
TBH we can get rid of `location` here as we can construct location as `base_dir` + `file.subpath()`

Reply:
I agree with the naming, but getting rid of the location IMO might be a bit too of an intrusive change, as we don't trivially concatenate the paths, but also perform mapping and validation on it as well, so doing it once and storing, like we are doing now seems like a better solution